### PR TITLE
Removed FromString and ToString proto conversion functions

### PR DESCRIPTION
**Pull request by mrice32**
They are no longer necessary for import.

### CODE REVIEW

**Comment by mrice32** [comment]
@htuch

File: source/common/protobuf/protobuf.h [hunk 1]
```diff
@@ -43,14 +43,9 @@ namespace ProtobufTypes {
...
 
-inline String ToString(const std::string& s) { return s; }
-inline std::string FromString(const String& s) { return s; }
-
 inline std::vector<std::string>
 stringVector(const Protobuf::RepeatedPtrField<ProtobufTypes::String>& source) {
-  std::vector<std::string> result;
-  std::transform(source.begin(), source.end(), std::back_inserter(result), FromString);
-  return result;
+  return std::vector<std::string>(source.begin(), source.end());
```
Comment (htuch) [inline source/common/protobuf/protobuf.h]: I'd just inline this and nuke the entire function.

Comment (mrice32) [inline source/common/protobuf/protobuf.h]: Okay.  Will do.

File: source/common/protobuf/utility.h [hunk 2]
```diff
@@ -41,7 +41,8 @@ class RepeatedPtrUtil {
 public:
   static std::string join(const Protobuf::RepeatedPtrField<ProtobufTypes::String>& source,
                           const std::string& delimiter) {
-    return StringUtil::join(ProtobufTypes::stringVector(source), delimiter);
+    std::vector<std::string> proto_strings(source.begin(), source.end());
```
Comment (mattklein123) [inline source/common/protobuf/utility.h]: nit: I would kill this local variable and just construct the vector temporary inline in the function call.

**Comment by mrice32** [comment]
Matt's nit should be fixed now.  I rebased because there were a few new additions of FromString and ToString in recent commits causing the CI to fail.

**Pull request merged**